### PR TITLE
Fold Travis CI output by component

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | 

Travis CI has [a feature that allows fold in tests output](https://github.com/travis-ci/travis-ci/issues/2285#issuecomment-42724719). For better tests readability, I propose to fold all components output that have successful result.

### CODE REVIEW

**Comment by stof** [comment]
wouldn't your change also fold in case of failure ?

**Comment by maidmaid** [comment]
Yes, but it's ``[WIP]``, I notify you when I have a good changes :)

**Comment by maidmaid** [comment]
@stof could you help me please? I want this result:
```
travis_fold:start:src.Symfony.Bridge.PhpUnit
[... composer update output ... ]
[... phpunit output ... ]
travis_fold:end:src.Symfony.Bridge.PhpUnit
```
but ``[... composer update output ... ]`` is the first output. I suspect ``parallel`` command. Do you have an idea?

**Comment by nicolas-grekas** [comment]
maybe redirect stderr to stdout?
also note that when you're read, this should be rebased on 2.7

**Comment by maidmaid** [comment]
Where is Travis CI? :confused:

**Comment by nicolas-grekas** [comment]
I guess you need to push force again (after a commit --amen)

**Comment by nicolas-grekas** [comment]
looks like its almost working :)
my 2cts: would be great to have two folds per component: one for composer, one for phpunit - and would be great to have failing folds be opened (either composer or phpunit, ie the one that is failing)

**Comment by maidmaid** [comment]
Ok @nicolas-grekas, I will see to do that :)

**Comment by maidmaid** [comment]
Here current result:

screenshot from 2017-04-06 13-04-00

WDYT?

**Comment by nicolas-grekas** [comment]
Looks nice!
what about a "fold" shell script in the `.github` folder, that would take a label as first arg, then a command ?

allowing someting like this to work (note that I removed folding for HHVM because I think we don't need it):
```sh
- if [[ ! $deps && ! $PHP = hhvm* ]]; then echo "$COMPONENTS" | parallel --gnu 'fold {} $PHPUNIT --exclude-group tty,benchmark,intl-data {}'; fi
- if [[ ! $deps && ! $PHP = hhvm* ]]; then fold PHPUnit.tty $PHPUNIT --group tty; fi
- if [[ ! $deps && $PHP = hhvm* ]]; then $PHPUNIT --exclude-group benchmark,intl-data; fi
- if [[ ! $deps && $PHP = ${MIN_PHP%.*} ]]; then echo -e "1\\n0" | xargs -I{} fold PHPUnit.sigchild sh -c 'ENHANCE_SIGCHLD={} php-$MIN_PHP/sapi/cli/php .phpunit/phpunit-4.8/phpunit --colors=always src/Symfony/Component/Process/'; fi
- if [[ $deps = high ]]; then echo "$COMPONENTS" | parallel --gnu -j10% fold {} sh -c 'cd {}; fold Composer composer update --no-progress --no-suggest --ansi && fold PHPUnit $PHPUNIT --exclude-group tty,benchmark,intl-data$LEGACY'; fi
- if [[ $deps = low ]]; then echo "$COMPONENTS" | parallel --gnu -j10% fold  {} sh -c 'cd {}; fold Composer composer update --no-progress --no-suggest --ansi --prefer-lowest --prefer-stable && fold PHPUnit $PHPUNIT --exclude-group tty,benchmark,intl-data'; fi
```

File: .travis.yml [hunk 1]
```diff
@@ -84,11 +84,12 @@ install:
     - if [[ ! $skip && ! $PHP = hhvm* ]]; then php -i; else hhvm --php -r 'print_r($_SERVER);print_r(ini_get_all());'; fi
 
 script:
-    - REPORT=' && echo -e "\\e[32mOK\\e[0m {}\\n\\n" || (echo -e "\\e[41mKO\\e[0m {}\\n\\n" && $(exit 1))'
+    - FOLD_START='echo "travis_fold:start:$FOLD"; echo -e "\\e[32m$FOLD\\e[0m"'
+    - FOLD_END='echo "travis_fold:end:$FOLD"'
```
Comment (nicolas-grekas) [inline .travis.yml]: such config lines should be moved in the previous "install" section

**Comment by maidmaid** [comment]
Ok for fold shell script. But in your code example, you have forget to end folding no? Instead:

```diff
- - if [[ ! $deps && ! $PHP = hhvm* ]]; then fold PHPUnit.tty $PHPUNIT --group tty; fi
+ - if [[ ! $deps && ! $PHP = hhvm* ]]; then fold PHPUnit.tty $PHPUNIT --group tty fold_end; fi
```

**Comment by nicolas-grekas** [comment]
There is no need for the end script if start/end magic string are handled inside the script in fact.

**Comment by maidmaid** [comment]
Resume of changes:
- fold shell script is created,
- shell function is renamed to ``tfold`` (to avoid collisions with ``fold`` unix command),
- output keeps alphabetic order (thanks to ``parallel -k``),
- fold title is in blue for better readability.

screenshot from 2017-04-13 21-14-02

**Comment by nicolas-grekas** [comment]
Thank you @maidmaid 
I took over in #22461 to add a rewrite of the .travis.yml file on top of your commit.

**Pull request closed without merging**